sdk/Sdk.Generators/FunctionMetadataProviderGenerator.cs

```diff
@@ -0,0 +1,384 @@
+                        indentedTextWriter.WriteLine(propertyName + " = " + arr + ",");
+                    }
+                    else
+                    {
+                        var propertyValue = FormatObject(prop.Value);
+                        indentedTextWriter.WriteLine(propertyName + " = " + propertyValue + ",");
+                    }
+                }
+
+                indentedTextWriter.Indent--;
+                indentedTextWriter.WriteLine("};");
+                indentedTextWriter.WriteLine("var " + functionName + "Binding" + bindingCount.ToString() + "JSONstring = JsonSerializer.Serialize(" + functionName + "Binding" + bindingCount.ToString() + ").ToString();");
+                indentedTextWriter.WriteLine(bindingsListName + ".Add(" + functionName + "Binding" + bindingCount.ToString() + "JSONstring);");
+
+                bindingCount++;
+
+                // auto-add a return type for http for now
+                if (string.Equals(bindingType, "httptrigger", StringComparison.OrdinalIgnoreCase))
+                {
+                    var totalBindings = method.ParameterList.Parameters.Count(param => param.AttributeLists.Count > 0);
+
+                    indentedTextWriter.WriteLine("var " + functionName + "Binding" + bindingCount.ToString() + " = new {");
+                    indentedTextWriter.Indent++;
+
+                    // if there are only two bindings and one is an httptrigger, then we use "$return" for the binding name
+                    if (totalBindings < 2)
+                    {
+                        indentedTextWriter.WriteLine("name = \"" + "$return" + "\",");
+                    }
+                    else
+                    {
+                        // when there are more than 2 bindings and one is an HttpTrigger, we can assume that there is another output binding
+                        // we return HttpResponse instead of "$return"
+                        indentedTextWriter.WriteLine("name = \"" + "HttpResponse" + "\",");
+                    }
+
+                    indentedTextWriter.WriteLine("type = \"" + "http" + "\",");
+                    indentedTextWriter.WriteLine("direction = \"" + "Out" + "\",");
+                    indentedTextWriter.Indent--;
+                    indentedTextWriter.WriteLine("};");
+                    indentedTextWriter.WriteLine("var " + functionName + "Binding" + bindingCount.ToString() + "JSONstring = JsonSerializer.Serialize(" + functionName + "Binding" + bindingCount.ToString() + ").ToString();");
+                    indentedTextWriter.WriteLine(functionName + "RawBindings.Add(" + functionName + "Binding" + bindingCount.ToString() + "JSONstring);");
+                    bindingCount++;
+                }
+            }
+        }
+
+        private static void AddEnumTypes(IndentedTextWriter indentedTextWriter)
+        {
+            indentedTextWriter.WriteLine("public enum AuthorizationLevel");
```

Comment:
If user function app has http trigger, will they end up with 2 instances of this enum definition? Does it produce a compilation error?

Reply:
There's no compilation error and this is only used to create gRpc defined function metadata and send to the host. It's a workaround because the source-gen api can't parse the enum type (it only provides numbers in its place like 0 for anonymous type). The parsing had to be written into the generated code but let me see if I can pull the type from the http extensions to do that parsing here.

Reply:
Trying it out and I think there's another issue - the user's app has to have a reference to the http extensions for us to use it. On the chance that their app doesn't, the source-generator would throw an error.